### PR TITLE
[soem] Remove MSVC /WX too, in addition to -Werror.

### PR DESCRIPTION
**Pull request by hmfhmf**
This PR adds to https://github.com/microsoft/vcpkg/pull/31889 by also removing the MSVC `/WX` flag, to disable "Treat all compiler warnings as errors" for MSVC too.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by hmfhmf** [comment]
> @hmfhmf please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Review by JonLiu1993 — APPROVED** [review]

File: versions/s-/soem.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "4bc716a66ac48f993a2242f2b584a897313901a3",
+      "version-date": "2023-06-09",
+      "port-version": 2
+    },
```
Comment (JonLiu1993) [inline versions/s-/soem.json:7]:
```suggestion
```

Comment (JonLiu1993) [reply versions/s-/soem.json:7]: Please remove this and run command "./vcpkg x--add-version soem --overwrite-version" and commit agagin.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: versions/s-/soem.json (hunk 1 above)
Comment (hmfhmf) [reply versions/s-/soem.json]: Thanks for helping out! I realized that the original mistake I made was to forget to change port-version in ports/soem/vcpkg.json before running `vcpkg x-add-version`. It's done now.

**Comment by JonLiu1993** [comment]
Tested usage successfully by `soem:x64-windows`:
````
soem provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(soem CONFIG REQUIRED)
  target_link_libraries(main PRIVATE soem)
````

**Review by JonLiu1993 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**